File: backend/app/schemas/report.py
from pydantic import BaseModel


class EmployeeReportRead(BaseModel):
    employee_id: int
    full_name: str
    position: str
    total_hours: float
    total_shifts: int


class EmployeeSelfReportRead(BaseModel):
    employee_id: int
    full_name: str
    total_hours: float
    total_shifts: int

File: mobile/android/app/src/main/java/com/froggyriia/shiftplanner/domain/model/Report.kt
package com.froggyriia.shiftplanner.domain.model

data class EmployeeReport(
    val employeeId: Int,
    val fullName: String,
    val position: String,
    val totalHours: Double,
    val totalShifts: Int
)

data class MySelfReport(
    val employeeId: Int,
    val fullName: String,
    val totalHours: Double,
    val totalShifts: Int
)
